src/go/serviceconfig/service_config_fetcher.go

```diff
@@ -19,32 +19,31 @@ import (
 	"crypto/x509"
 	"fmt"
 	"io/ioutil"
-	"math"
 	"net/http"
 	"time"
 
-	"github.com/GoogleCloudPlatform/esp-v2/src/go/metadata"
 	"github.com/GoogleCloudPlatform/esp-v2/src/go/options"
 	"github.com/GoogleCloudPlatform/esp-v2/src/go/util"
 	"github.com/golang/glog"
 	"github.com/golang/protobuf/proto"
 
 	confpb "google.golang.org/genproto/googleapis/api/serviceconfig"
-	smpb "google.golang.org/genproto/googleapis/api/servicemanagement/v1"
 )
 
 type ServiceConfigFetcher struct {
 	serviceName         string
 	checkRolloutsTicker *time.Ticker
 	client              http.Client
-	mf                  *metadata.MetadataFetcher
-	opts                options.ConfigGeneratorOptions
+	opts                *options.ConfigGeneratorOptions
+
+	accessToken func() (string, time.Duration, error)
```

Comment:
Would prefer if name clarifies the need for an async call:

```
accessTokenGetter? accessTokenFunc?
configIdGetter? configIdFunc?
```

Reply:
Per https://golang.org/doc/effective_go.html#Getters
Getter is just the name of variable